Index: src/mac/carbon/settings.cpp
===================================================================
--- src/mac/carbon/settings.cpp	(revision 52547)
+++ src/mac/carbon/settings.cpp	(working copy)
@@ -118,7 +118,9 @@
             break ;
 
         case wxSYS_COLOUR_INFOBK :
-            resultColor = *wxWHITE ;
+            // we don't have a way to detect tooltip color, so use the
+            // standard value used at least on 10.4:
+            resultColor = wxColour( 0xFF, 0xFF, 0xD3 ) ;
             break ;
         case wxSYS_COLOUR_APPWORKSPACE:
             resultColor =  wxColor( 0x80, 0x80, 0x80 ); ;
