diff -ru wxWidgets-2.8.7.orig/include/wx/mac/carbon/listctrl.h wxWidgets-2.8.7/include/wx/mac/carbon/listctrl.h
--- wxWidgets-2.8.7.orig/include/wx/mac/carbon/listctrl.h	2008-06-07 02:02:07.000000000 +0200
+++ wxWidgets-2.8.7/include/wx/mac/carbon/listctrl.h	2008-06-07 02:03:07.000000000 +0200
@@ -419,6 +419,9 @@
                                   // keep track of inserted/deleted columns
 
   int               m_count; // for virtual lists, store item count
+
+private:
+  int CalcColumnAutoWidth(int col) const;
   
 private: 
   DECLARE_EVENT_TABLE()
diff -ru wxWidgets-2.8.7.orig/src/mac/carbon/listctrl_mac.cpp wxWidgets-2.8.7/src/mac/carbon/listctrl_mac.cpp
--- wxWidgets-2.8.7.orig/src/mac/carbon/listctrl_mac.cpp	2008-06-07 02:02:07.000000000 +0200
+++ wxWidgets-2.8.7/src/mac/carbon/listctrl_mac.cpp	2008-06-07 02:04:28.000000000 +0200
@@ -1051,9 +1051,10 @@
 
     if (m_dbImpl)
     {
-        int mywidth = width;
-        if (width == wxLIST_AUTOSIZE || width == wxLIST_AUTOSIZE_USEHEADER)
-            mywidth = 150;
+        if ( width == wxLIST_AUTOSIZE_USEHEADER )
+        {
+            width = 150; // FIXME
+        }
 
         if (col == -1)
         {
@@ -1065,17 +1066,22 @@
                 colInfo.SetWidth(width);
                 SetColumn(col, colInfo);
 
+                const int mywidth = (width == wxLIST_AUTOSIZE)
+                                    ? CalcColumnAutoWidth(column) : width;
                 m_dbImpl->SetColumnWidth(col, mywidth);
             }
         }
         else
         {
+            if ( width == wxLIST_AUTOSIZE )
+                width = CalcColumnAutoWidth(col);
+
             wxListItem colInfo;
             GetColumn(col, colInfo);
             
             colInfo.SetWidth(width);
             SetColumn(col, colInfo);
-            m_dbImpl->SetColumnWidth(col, mywidth);
+            m_dbImpl->SetColumnWidth(col, width);
         }
         return true;
     }
@@ -3338,5 +3344,36 @@
     }
 }
 
+int wxListCtrl::CalcColumnAutoWidth(int col) const
+{
+    int width = 0;
+
+    for ( int i = 0; i < GetItemCount(); i++ )
+    {
+        wxListItem info;
+        info.SetMask(wxLIST_MASK_TEXT | wxLIST_MASK_IMAGE);
+        info.SetId(i);
+        info.SetColumn(col);
+        GetItem(info);
+
+        const wxFont font = info.GetFont();
+
+        int w = 0;
+        if ( font.IsOk() )
+            GetTextExtent(info.GetText(), &w, NULL, NULL, NULL, &font);
+        else
+            GetTextExtent(info.GetText(), &w, NULL);
+
+        w += 2 * kItemPadding;
+
+        if ( info.GetImage() != -1 )
+            w += kIconWidth;
+
+        width = wxMax(width, w);
+    }
+
+    return width;
+}
+
 #endif // wxUSE_LISTCTRL
 
