--- unistd.in.h.orig	2013-02-25 18:40:16.000000000 +0900
+++ unistd.in.h	2013-02-25 18:49:26.754601400 +0900
@@ -109,7 +109,11 @@
 
 /* Get getopt(), optarg, optind, opterr, optopt.
    But avoid namespace pollution on glibc systems.  */
-#if @GNULIB_UNISTD_H_GETOPT@ && !defined __GLIBC__ && !defined _GL_SYSTEM_GETOPT
+/* Also, don't include <getopt.h> inside libgettextpo, because we use
+   the getopt module only in gettext-tools/gnulib-lib/, not in
+   gettext-tools/libgettextpo/, but there is only a single
+   GNULIB_UNISTD_H_GETOPT variable for both.  */
+#if @GNULIB_UNISTD_H_GETOPT@ && !defined GTPO_CONFIG_H && !defined __GLIBC__ && !defined _GL_SYSTEM_GETOPT
 # define __need_getopt
 # include <getopt.h>
 #endif
