diff -ru gettext-0.26.orig/gettext-runtime/Makefile.am gettext-0.26/gettext-runtime/Makefile.am
--- gettext-0.26.orig/gettext-runtime/Makefile.am	2025-06-25 04:16:03
+++ gettext-0.26/gettext-runtime/Makefile.am	2025-07-30 20:08:36
@@ -26,7 +26,7 @@
 else
 SUBDIR_libasprintf =
 endif
-SUBDIRS = doc intl intl-java intl-csharp intl-d intl-modula2 gnulib-lib $(SUBDIR_libasprintf) src po man m4 tests install-tests
+SUBDIRS = doc intl intl-java intl-csharp intl-d intl-modula2 gnulib-lib $(SUBDIR_libasprintf) src po m4
 
 EXTRA_DIST = BUGS
 
diff -ru gettext-0.26.orig/gettext-tools/Makefile.am gettext-0.26/gettext-tools/Makefile.am
--- gettext-0.26.orig/gettext-tools/Makefile.am	2025-06-19 11:49:30
+++ gettext-0.26/gettext-tools/Makefile.am	2025-07-30 20:08:09
@@ -19,7 +19,7 @@
 AUTOMAKE_OPTIONS = 1.5 gnu
 ACLOCAL_AMFLAGS = -I m4 -I ../gettext-runtime/m4 -I ../m4 -I gnulib-m4 -I libgrep/gnulib-m4 -I libgettextpo/gnulib-m4 -I tests/gnulib-m4
 
-SUBDIRS = gnulib-lib libgrep src libgettextpo po its projects styles emacs misc man m4 tests system-tests gnulib-tests install-tests examples doc
+SUBDIRS = gnulib-lib libgrep src libgettextpo po its projects styles emacs misc m4
 
 EXTRA_DIST = misc/disclaim-translations.txt
 MOSTLYCLEANFILES = core *.stackdump
