*** modules/regex.orig	2013-01-03 10:18:51.000000000 +0900
--- modules/regex	2013-01-03 10:20:47.000000000 +0900
***************
*** 18,24 ****
  alloca-opt      [test $ac_use_included_regex = yes]
  btowc           [test $ac_use_included_regex = yes]
  gettext-h       [test $ac_use_included_regex = yes]
! malloc-gnu      [test $ac_use_included_regex = yes]
  memcmp          [test $ac_use_included_regex = yes]
  memmove         [test $ac_use_included_regex = yes]
  mbrtowc         [test $ac_use_included_regex = yes]
--- 18,24 ----
  alloca-opt      [test $ac_use_included_regex = yes]
  btowc           [test $ac_use_included_regex = yes]
  gettext-h       [test $ac_use_included_regex = yes]
! malloc-posix    [test $ac_use_included_regex = yes]
  memcmp          [test $ac_use_included_regex = yes]
  memmove         [test $ac_use_included_regex = yes]
  mbrtowc         [test $ac_use_included_regex = yes]
